src/App.tsx

```diff
@@ -1,44 +1,195 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/bulma.sass';
 import '@fortawesome/fontawesome-free/css/all.css';
 import './App.scss';
-
 import classNames from 'classnames';
+
+import { User } from './types/User';
+import { Post } from './types/Post';
+import { Comment } from './types/Comment';
+import { Reset } from './types/Reset';
+
+import { getUsersList } from './api/users';
+import { getPosts } from './api/posts';
+import { getComments, postComment, deleteComment } from './api/comments';
+
 import { PostsList } from './components/PostsList';
 import { PostDetails } from './components/PostDetails';
 import { UserSelector } from './components/UserSelector';
 import { Loader } from './components/Loader';
 
 export const App: React.FC = () => {
+  const [userId, setUserId] = useState<number | null>(null);
+  const [userName, setUsername] = useState('Choose a user');
+  const [users, setUsers] = useState<User[] | null>(null);
+  const [posts, setPosts] = useState<Post [] | null>(null);
+  const [isPostLoading, setIsPostLoading] = useState(false);
+  const [isWrongPostResponse, setIsWrongPostResponse] = useState(false);
+  const [isActiveList, setIsActiveList] = useState(false);
+  const [selectedPost, setSelectedPost] = useState<Post | null>(null);
+  const [comments, setComments] = useState<Comment[] | null>(null);
+  const [isWrongSidebarResponse, setIsWrongSidebarResponse] = useState(false);
+  const [isCommentsLoading, setIsCommentsLoading] = useState(false);
+  const [isCommentForm, setIsCommentForm] = useState(false);
+  const [isLoadingNewComment, setIsloadingNewComment] = useState(false);
+  const [isDangerSubmit, setIsDangerSubmit] = useState(false);
+  const [isWarningUpdate, setIsWarningUpdate] = useState(false);
+
+  useEffect(() => {
+    getUsersList()
+      .then(response => setUsers(response))
+      .catch(error => window.alert(error));
```

Comment:
don't use an alert to show an error. Either console.error it or throw a new error